src/vehicle.cpp

```diff
@@ -225,6 +225,18 @@ vehicle::~vehicle()
 {
 }
 
+void vehicle::set_hp( vehicle_part &pt, int qty )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    pt.base.set_damage( pt.base.max_damage() - ( qty * k ) );
+}
+
+bool vehicle::mod_hp( vehicle_part &pt, int qty, damage_type dt )
+{
+    double k = pt.base.max_damage() / double( pt.info().durability );
+    return pt.base.mod_damage( - qty * k, dt );
```

Comment:
Then it's a tangled logic that doesn't seem to be consistent with anything. At the very least you could extract this part into a separate function:
```c++
double k = pt.base.max_damage() / double( pt.info().durability );
- qty * k
```
Also, if `pt.info().durability` is zero then you get a `NaN` which cannot be converted into `int` (undefined behavior).